engines/freescape/language/8bitDetokeniser.cpp

```diff
@@ -0,0 +1,414 @@
+			conditionalInstructions->push_back(currentInstruction);
+			currentInstruction = FCLInstruction(Token::UNKNOWN);
+			break;
+		case 27:
+			detokenisedStream += "DELAY (";
+			currentInstruction = FCLInstruction(Token::DELAY);
+			break;
+		case 28:
+			detokenisedStream += "SYNCSND (";
+			currentInstruction = FCLInstruction(Token::SOUND);
+			currentInstruction.setAdditional(true);
+			break;
+		case 29:
+			detokenisedStream += "TOGGLEBIT (";
+			currentInstruction = FCLInstruction(Token::TOGGLEBIT);
+			break;
+
+		case 25:
+			// this should toggle border colour or the room palette
+			detokenisedStream += "SPFX (";
+			break;
+
+		case 20:
+			detokenisedStream += "SETVAR ";
+			detokenisedStream += Common::String::format("(%d, v%d)", (int)tokenisedCondition[bytePointer], (int)tokenisedCondition[bytePointer + 1]);
+			bytePointer += 2;
+			numberOfArguments = 0;
+			break;
+
+		case 35:
+			detokenisedStream += "SCREEN (";
+			break;
+
+		case 44:
+			detokenisedStream += "ELSE ";
+			break;
+
+		case 45:
+			detokenisedStream += "ENDIF ";
+			break;
+
+		case 46:
+			detokenisedStream += "IFGTE ";
+			detokenisedStream += Common::String::format("(v%d, %d)", (int)tokenisedCondition[bytePointer], (int)tokenisedCondition[bytePointer + 1]);
+			bytePointer += 2;
+			numberOfArguments = 0;
+			break;
+
+		case 47:
+			detokenisedStream += "IFLTE ";
+			detokenisedStream += Common::String::format("(v%d, %d)", (int)tokenisedCondition[bytePointer], (int)tokenisedCondition[bytePointer + 1]);
+			bytePointer += 2;
+			numberOfArguments = 0;
+			break;
+
+		case 48:
+			detokenisedStream += "EXECUTE ";
+			break;
+		}
+
+		// if there are any regular arguments to add, do so
+		if (numberOfArguments) {
+			for (uint8 argumentNumber = 0; argumentNumber < numberOfArguments; argumentNumber++) {
+				if (argumentNumber == 0)
+					currentInstruction.setSource(tokenisedCondition[bytePointer]);
+				else if (argumentNumber == 1)
+					currentInstruction.setDestination(tokenisedCondition[bytePointer]);
+				else
+					error("Unexpected number of arguments!");
+
+				detokenisedStream += Common::String::format("%d", (int)tokenisedCondition[bytePointer]);
+				bytePointer++;
+
+				if (argumentNumber < numberOfArguments - 1)
+					detokenisedStream += ", ";
+			}
+
+			detokenisedStream += ")";
+			conditionalInstructions->push_back(currentInstruction);
+			currentInstruction = FCLInstruction(Token::UNKNOWN);
+		}
+
+		// throw in a newline
+		detokenisedStream += "\n";
+	}
+
+	// if (!conditionalInstructions)
+	//	conditionalInstructions = new FCLInstructionVector();
+
+	// conditionalInstructions->push_back(currentInstruction);
+
+	FCLInstruction branch;
+	if (conditionalIsShot)
+		branch = FCLInstruction(Token::SHOTQ);
+	else
+		branch = FCLInstruction(Token::COLLIDEDQ);
+
+	branch.setBranches(conditionalInstructions, nullptr);
+	instructions.push_back(branch);
+
+	return (new Common::String(detokenisedStream));
```

Comment:
This tends to leak memory in the user code. Use `return Common::String(detokenisedStream)`

Reply:
Fixed